sdk/go/apigateway/restAPI.go

```diff
@@ -114,7 +114,7 @@ type RestAPIArgs struct {
 	// specification for the API.
 	//
 	// Either `swaggerString` or `routes` must be specified.
-	Routes []RouteArgs
+	Routes []RouteInput
```

Comment:
Unclear why switching to `pulumi package gen-sdk` causing this argument to change. This appears to be a breaking change as I also had to update the test code to make it work after this upgrade. It seems that it is marked as plain in the schema, so it should not be an input interface:

https://github.com/pulumi/pulumi-aws-apigateway/blob/b65b7317fe32326ae1caf19edb7e96e6e795dc41/schema.yaml#L357-L362

cc: @Frassle

Reply:
This is definitely a breaking change. Since we see this only in the Go SDK, I strongly suspect it's a Go codegen bug.

Reply:
Yeah let's isolate the update to new codegen into a separate PR and bisect this to get this in. We can check this PR on top with a clean diff.

Reply:
I can help look tomorrow since I'm on AWS ops.

Reply:
This _might_ be a schema marshalling bug that caused the change, but yes probably a Go SDK bug as well if only Go changed due to it.

Reply:
The old code was referencing 3.107.0

This originally upgraded to latest (3.126.0)

The earliest version I can downgrade to is 3.113.2 because of needing the version argument, this also exhibits the same behaviour.

Anton points out in another comment that this does also affect nodejs and python (but not dotnet and java) though this isn't a breaking change in typescript and python.

Reply:
Just to rule it out, converting the schame to JSON instead of YAML to match other providers doesn't have any affect, so it's not related to the YAML marshalling.

Reply:
I've reverted back to using the custom program for now but we can undo the revert once https://github.com/pulumi/pulumi/pull/16812 has been released.